appinventor/appengine/src/com/google/appinventor/client/explorer/project/ProjectManager.java

```diff
@@ -29,6 +31,8 @@ public final class ProjectManager {
   // List of listeners for any project manager events.
   private final List<ProjectManagerEventListener> projectManagerEventListeners;
 
+  private Promise<List<Project>> loadProjectPromise = null;
```

Comment:
@ewpatton I don't believe this should be an instance variable. It means that if a project is added, the client will never learn of it because it will use the values fetched from some prior point in time, possibly before a project was added. I suspect this is the cause of the behavior I reported in my previous comment.

Reply:
I will double check this, but my logic here is that we only ever load the project list once from the server. This promise represents the status of that call. Subsequent adding/removing of projects within a session take a different path. It likely makes sense to rename the variable to clear up any confusion.

Reply:
It's not the only call. I'm on my phone, so cannot check the reference.

Reply:
@ewpatton So loadProjects is called from showWelcomeDialog which is called by ShowSplashAction which is called when the user does Help->Show Splash Screen. And this can happen after a new project has been created.